core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -231,24 +231,34 @@ abstract class RawModule extends BaseModule {
       .getOrElse(
         throwException(s"BoringUtils currently only support identity views, ${_right} has multiple targets.")
       )
-    val rhs = (left.probeInfo.nonEmpty, right.probeInfo.nonEmpty) match {
-      case (true, true)                                 => ProbeDefine(si, left.lref, Node(right))
-      case (true, false) if left.probeInfo.get.writable => ProbeDefine(si, left.lref, RWProbeExpr(Node(right)))
-      case (true, false)                                => ProbeDefine(si, left.lref, ProbeExpr(Node(right)))
-      case (false, true)                                => Connect(si, left.lref, ProbeRead(Node(right)))
-      case (false, false)                               =>
-        // For non-probe, directly create Nodes for lhs, skipping visibility check to support BoringUtils.drive.
-        (left, right) match {
-          case (_: Property[_], _: Property[_]) => PropAssign(si, Node(left), Node(right))
-          // Use `connect lhs, read(probe(rhs))` if lhs is passive version of rhs.
-          // This provides solution for this: https://github.com/chipsalliance/chisel/issues/3557
-          case (_, _)
-              if !DataMirror.checkAlignmentTypeEquivalence(left, right) &&
-                DataMirror.checkAlignmentTypeEquivalence(left, Output(chiselTypeOf(right))) =>
-            Connect(si, Node(left), ProbeRead(ProbeExpr(Node(right))))
-          case (_, _) => Connect(si, Node(left), Node(right))
-        }
+
+    def computeConnection(left: Data, right: Data, si: SourceInfo): Command = {
```

Comment:
```suggestion
    def computeConnection(left: Data, right: Data): Command = {
```
Just delete this argument, you already capture `si` above since this is an inner method. The compilation error in CI is due to the fact that `lref` needs the implicit argument and by shadowing the name of the variable, it makes the `implicit si` above not available (lol, fixed by not needing to name `using` parameters in Scala 3)!

Reply:
Oops, I didn't notice that in my local test somehow.